### DISCUSSION TITLE
Please  HEEEEEEELP （T_T）

### DISCUSSION THREAD

**Discussion by Yina1029** [jamesjiadazhan/dietaryindex#279] [Q&A] [answered]
Hi, Dr. Zhan, I saw that the food intake in the PHDI code you gave is in grams, while in the FPED file it is in non-grams. What code conversion could I use? Using R is  the best.

**Comment by jamesjiadazhan** [comment]
Sorry this is not supported for NHANES.

**Reply by Yina1029** [reply]
so……you mean that the FPED data cannot be used to calculate the PHDI with the code you gave, right?

**Reply by muzi18768163385-create** [reply]
Does PHDI use the food intake calculation in the FPED file?

**Reply by jamesjiadazhan** [reply]
Please check this code and data repository. PHDI can be calcualted in NHANES but it has to be done in a complex way since all food groups need to be in grams. https://zenodo.org/records/18042895

**Comment by muzi18768163385-create** [comment]
We already have FPD data, all FPD data from 2005 to 2020. Can I directly use this data to convert and calculate PHDI? Can I use this conversion formula to directly convert to grams and then calculate PHDI? 1 ounce (oz) = 28.3495 grams (g); 1 cup of juice = 240 grams; 1 teaspoon (tsp) of added sugar = 4.2 grams.

**Comment by muzi18768163385-create** [comment]
I mean,can we use the FPED data from USDA TO calculate the PHDI? Use the following formula 
-       1 oz = 28.3495 g
-	1 cup fruit juice = 240 g
-	1 tsp table sugar = 4.2 g
1 oz eq grains	=30 g| 
| 1 oz eq meat/fish |=28–30 g** ||
 1 cup fruit |=150 g** |
| 1 cup vegetables | =130 g
1 cup milk/yogurt=245 g
to estimate them into grams

**Reply by jamesjiadazhan** [reply, marked as answer]
This is a naive method, which leads to inaccurate results. Different foods have different conversion factors. Not all the fruits is 1 cup = 150 g. For more details, see this paper:
- Zhan J, Bui L, Hodge RA, et al. Planetary Health Diet Index Trends and Associations with Dietary Greenhouse Gas Emissions, Disease Biomarkers, Obesity, and Mortality in the United States (2005-2018). Am J Clin Nutr. 2025;121(3):580-588. doi:10.1016/j.ajcnut.2025.01.007

**Reply by muzi18768163385-create** [reply]
Thank you so much!!!
